lib/send_assets.js

```diff
@@ -66,35 +83,62 @@ const SendAssets = {
         const amount = getAmount(data.Amount, decimals)
         console.debug('new amount: ' + amount)
         console.debug({ toAddress })
-        const outerTxns = await this.getMailboxSendTransactions(algodClient, assetId, 
-          amount, fromAccount, toAddress)
+        const outerTxns = await this.getMailboxSendTransactions(
+          algodClient,
+          assetId,
+          amount,
+          fromAccount,
+          toAddress
+        )
         if (outerTxns.length == 1) {
           simpleAssetTransferCount++
         }
         console.debug({ outerTxns })
-        helper.packageUnsignedTransactions(allTxns, txnsForSigning, k, outerTxns)
+        helper.packageUnsignedTransactions(
+          allTxns,
+          txnsForSigning,
+          k,
+          outerTxns
+        )
       }
 
       // Add operator fee transaction for direct transfers of asset
       if (simpleAssetTransferCount > 0) {
         const operatorFee = simpleAssetTransferCount * 50000 // 0.05 algo fee to operator per txn
-        const outerSendTxn = await generateTransactions.getSimpleSendAlgosTransaction(algodClient, 
-          operatorFee, fromAccount, feeReceiverAddress)
-        helper.packageUnsignedTransactions(allTxns, txnsForSigning, toData.length, outerSendTxn)
+        const outerSendTxn =
+          await generateTransactions.getSimpleSendAlgosTransaction(
+            algodClient,
+            operatorFee,
+            fromAccount,
+            feeReceiverAddress
+          )
+        helper.packageUnsignedTransactions(
+          allTxns,
+          txnsForSigning,
+          toData.length,
+          outerSendTxn
+        )
       }
+      events.emit('sending-assets-event', { status: 'signing1' })
+      const signedAllTxns = await helper.signWithMyAlgo(
+        txnsForSigning,
+        allTxns
+      )
 
-      const signedAllTxns = await helper.signWithMyAlgo(txnsForSigning, allTxns)
-
-      const confirmedTransactions = await helper.sendAllTransactions(signedAllTxns)
+      const confirmedTransactions = await helper.sendAllTransactions(
+        signedAllTxns
+      )
 
       let transResults = JSON.stringify(confirmedTransactions, null, 2)
       console.debug('trans results after confirmed are: ')
       console.debug(transResults)
+      events.emit('sending-assets-event', { status: 'complete' })
```

Comment:
- [ ] Unecissary event, needs to be inside of `sendAllTransactions` to emit the actual status of sending transactions. The event can be called `send-all-transactions` and it should include the `total` and `progress` values in the payload